core/domain/collection_domain.py

```diff
@@ -1019,6 +1019,26 @@ def _recursively_find_learned_skills(node):
         else:
             return []
 
+    def get_nodes_in_playable_order(self):
+        """Returns a list of collection node_dict in linear, playable order and
+        assumes the nodes can fit a linear structure.
+
+        Returns:
+            list(dict). A sorted list of collection nodes in dict.
+        """
+        sorted_exp_ids = self.init_exploration_ids
+        next_exp_ids = self.get_next_exploration_ids(sorted_exp_ids)
+        while next_exp_ids:
+            for next_exp_id in next_exp_ids:
+                if next_exp_id not in sorted_exp_ids:
+                    sorted_exp_ids.append(next_exp_id)
+            next_exp_ids = self.get_next_exploration_ids(sorted_exp_ids)
+
+        sorted_node_dicts = [
+            self.get_node(exp_id).to_dict() for exp_id in sorted_exp_ids]
+
+        return sorted_node_dicts
```

Comment:
In general, we usually only return dicts from to_dict or in special circumstances (like migrations). For consistency, should this method return nodes instead of dicts?

Reply:
Done!